AntistasiOfficial.Altis/resourcecheck.sqf

```diff
@@ -146,6 +152,31 @@ while {true} do {
 		};
 	} forEach ciudades;
 
+	if(countSupplyCrates < 6) then
+	{
+		_cityDecreased = false;
+		for "_i" from 0 to 4 do
+		{
+			_currentCity = selectRandom _cityInRange;
+			_types = [_currentCity, "GOOD"] call AS_fnc_getHighSupplies;
+			if(random 100 < 10) then {_types = [_currentCity, "LOW"] call AS_fnc_getHighSupplies};
+			if (((count _types) != 0) AND !_cityDecreased) then
+			{
+				_cityDecreased = true;
+				_type = selectRandom _types;
```

Comment:
_type return "GOOD" which is a string we can't pass into AS_fnc_changeCitySypply at line 169 below causing an error

Reply:
Yes we can.
The logic would be that sometimes in AS_fnc_changeCitySypply we receive other value than these 3 so _index will not be defined so before these conditions we add:

```
_index = -1

if(_type == "FOOD") then {_index = 0;};
if(_type == "WATER") then {_index = 1;};
if(_type == "FUEL") then {_index = 2;};

if (_index < 0) then { exit();};
```

Reply:
well, but if you pass "GOOD" it's not going ot match any of the logcal type below as you said

Reply:
Then add the support of the GOOD choice.
`if(_type == "GOOD") then {_index = 3;};`
and further in the file define the thing you want to do when we receive this GOOD value.

Reply:
Fixed